src/component-library/Tabs/Tabs.tsx

```diff
@@ -0,0 +1,71 @@
+import { useFocusRing } from '@react-aria/focus';
+import { useTabList } from '@react-aria/tabs';
+import { mergeProps } from '@react-aria/utils';
+import { useTabListState } from '@react-stately/tabs';
+import { CollectionChildren } from '@react-types/shared';
+import { forwardRef, HTMLAttributes, Key, useEffect, useState } from 'react';
+
+import { useDOMRef } from '../utils/dom';
+import { Tab } from './Tab';
+import { TabPanel } from './TabPanel';
+import { TabList, TabListWrapper, TabSelection } from './Tabs.style';
+
+type Props = {
+  defaultSelectedKey?: Key;
+  selectedKey?: Key;
+  onSelectionChange?: (index: Key) => void;
+  disabledKeys?: Key[];
+};
+
+type NativeAttrs = Omit<HTMLAttributes<unknown>, keyof Props>;
+
+type TabsProps = Props & NativeAttrs;
+
+const Tabs = forwardRef<HTMLDivElement, TabsProps>(
+  ({ children, ...props }, ref): JSX.Element => {
+    // eslint-disable-next-line @typescript-eslint/ban-types
+    const ariaProps = { children: children as CollectionChildren<object>, ...props };
```

Comment:
Their hooks are strongly typed to receive children of type `CollectionChildren`. This type does not allow to do conditional rendering on a TabItem.

```
...
{isLoading && <TabsItem title="staking">Staking</ TabsItem>}
...
```

This creates a type error. There is a reason for that ofc: [link](https://react-spectrum.adobe.com/react-stately/Collection.html)

Reply:
Unless there's a good reason to keep the `@typescript-eslint/ban-types` rule, I'd prefer to turn it off in the config rather than use disable-next-line

Reply:
fix lint rule

Reply:
As discussed, this is fine as it is for now - we'll take a look at it again in the next couple of weeks, and decided whether or not disable the rule

Reply:
@tomjeatt we can avoid the disable just be using a `object` alternative: `Record<string, unknown>` (found this on ts docs)